### PR TITLE
added --revision flag to `service deploy` cmd

### LINKED ISSUE #28: Deploy task definition revisions to services
It would be cool to add the ability to deploy a particular task definition revision to a service.  This would provide true 12-factor rollbacks where a previous docker image and environment variable configuration could be re-deployed exactly as it was.

Particularly now that we have:
```
REVISION=$(fargate task register -i <image> -e <KEY=VAL>)
```

example:
```
fargate service deploy --revision 42
```

### PR DESCRIPTION
**Pull request by ajliv**
Allows deploying a task definition revision to a service.

Closes #28 

```
Deploy applications to services

The Docker container image to use in the service can be specified
via the --image flag.

The docker-compose.yml format is also supported using the --file flag.
If -f is specified, the image and the environment variables in the
docker-compose.yml file will be deployed.

A task definition revision can be specified via the --revision flag.
The revision number can either be absolute or a delta specified with a sign
such as +5 or -2, where -2 is "2 configurations ago" from the current
deployed revision.

Usage:
  fargate service deploy [flags]

Examples:

fargate service deploy -i 123456789.dkr.ecr.us-east-1.amazonaws.com/my-service:1.0
fargate service deploy -f docker-compose.yml
fargate service deploy -r 37


Flags:
  -f, --file string       Specify a docker-compose.yml file to deploy. The image and environment variables in the file will be deployed.
  -h, --help              help for deploy
  -i, --image string      Docker image to run in the service
      --image-only        Only deploy the image when a docker-compose.yml file is specified.
  -r, --revision string   Task definition revision number

Global Flags:
  -c, --cluster string   ECS cluster name
      --no-emoji         Disable emoji output
      --nocolor          Disable color output
      --region string    AWS region (default "us-east-1")
  -s, --service string   ECS service name
  -v, --verbose          Verbose output
```

### CODE REVIEW

**Comment by ajliv** [comment]
Right now the code has the priority go to the `-f` flag, then `-r`, and falls back to `-i`. I guess a warning could be added for mixing incompatible flags. Maybe outside of the scope of the PR?

**Comment by jritsema** [comment]
Maybe something like this...https://github.com/turnerlabs/fargate/blob/develop/cmd/task_register.go#L39-L45

File: ecs/task_definition.go [hunk 1]
```diff
@@ -291,3 +292,39 @@ func (ecs *ECS) GetCpuAndMemoryFromTaskDefinition(taskDefinitionArn string) (str
 
 	return aws.StringValue(taskDefinition.Cpu), aws.StringValue(taskDefinition.Memory)
 }
+
+//ResolveRevisionNumber returns a task defintion revision number by absolute value or expression
+func (ecs *ECS) ResolveRevisionNumber(taskDefinitionArn string, revisionExpression string) string {
```
Comment (jritsema) [inline ecs/task_definition.go]: Seems like a good thing to write a test for :)

File: cmd/service_deploy.go [hunk 2]
```diff
@@ -31,24 +36,41 @@ via the --image flag.
...
 `,
 	Example: `
 fargate service deploy -i 123456789.dkr.ecr.us-east-1.amazonaws.com/my-service:1.0
 fargate service deploy -f docker-compose.yml
+fargate service deploy -r 37
 `,
 	Run: func(cmd *cobra.Command, args []string) {
 		operation := &ServiceDeployOperation{
 			ServiceName: getServiceName(),
+			Task:        getTaskName(),
```
Comment (jritsema) [inline cmd/service_deploy.go]: I don't think task should be required to deploy -i or -f (breaks backcompat).  If you don't specify it you get...
```shell
$ fargate service deploy -f docker-compose.yml
please specify task family using: fargate.yml, FARGATE_TASK envvar, or --task
```

Comment (ajliv) [reply cmd/service_deploy.go]: Good catch, fixed to grab the task family from the current task ARN

**Pull request merged**